### ISSUE TITLE
Improve FAQ re: Docker for Windows and Windows/Linux containers option

### ISSUE THREAD

**Issue by londoncalling** [docker/docker.github.io#4277]
### Problem description

Some feed back on your FAQs page based on work we have been doing, specifically implementing a Linux based container on Windows Server 2016 (containers which run very successfully on Windows 10) .

Your FAQs page at: https://docs.docker.com/engine/faq/ states that:
Does Docker run on Linux, macOS, or Windows?
The Docker Engine client runs natively on ... Windows. By default, these clients ... provide the required features to run Linux-based containers within ... Windows, or Windows-based containers on Windows.

It this 100% correct?  Is it 100% clear? 

What's that "or" doing there and what does it imply ... that we can run either Linux based containers OR Windows-based-containers on Windows? Should this be an "and"?

Specifically, what about Linux containers running on Windows Server 2016? 
At best, this does not appear to be straightforward ... an indication to this effect (assuming it is true?) and a link to recommended approaches would be very valuable.

In summary, there seems to be a significant difference between the implementations for Windows-10 and Windows Server 2016 that I think is not clear in the FAQs as they stand, and that could be usefully added.

### Related

@friism @ncbateman filed this issue as an email

**Comment by mstanleyjones** [comment]
Duplicate of #4264
